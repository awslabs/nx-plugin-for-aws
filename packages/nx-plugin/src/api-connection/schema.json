{
  "$schema": "https://json-schema.org/schema",
  "$id": "APIConnection",
  "title": "",
  "type": "object",
  "properties": {
    "sourceProject": {
      "type": "string",
      "description": "The source project which will call the API",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "Select the source project which will call the API",
      "x-dropdown": "projects"
    },
    "targetProject": {
      "type": "string",
      "description": "The target project containing your API",
      "x-prompt": "Select the target project containing your API",
      "x-dropdown": "projects"
    }
  },
  "required": ["sourceProject", "targetProject"]
}
