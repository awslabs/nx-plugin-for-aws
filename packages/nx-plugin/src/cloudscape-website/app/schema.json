{
  "$schema": "https://json-schema.org/schema",
  "cli": "nx",
  "$id": "Cloudscape App",
  "title": "Create a Cloudscape React Application",
  "description": "Create a Cloudscape React application for Nx.",
  "examples": [
    {
      "command": "nx g app myapp --directory=myorg",
      "description": "Generate `apps/myorg/myapp` and `apps/myorg/myapp-e2e`"
    }
  ],
  "type": "object",
  "properties": {
    "name": {
      "description": "The name of the application.",
      "type": "string",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use for the application?",
      "pattern": "^[a-zA-Z][^:]*$"
    },
    "directory": {
      "description": "The directory of the new application.",
      "type": "string",
      "alias": "dir",
      "x-priority": "important",
      "default": "packages",
      "x-prompt": "What directory would you like to store your application in?"
    }
  },
  "required": ["name"],
  "examplesFile": "../../../docs/application-examples.md"
}
