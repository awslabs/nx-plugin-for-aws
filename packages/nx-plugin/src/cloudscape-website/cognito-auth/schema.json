{
  "$schema": "https://json-schema.org/schema",
  "cli": "nx",
  "$id": "Cloudscape App",
  "title": "Create a Cloudscape React Application",
  "description": "Create a Cloudscape React application for Nx.",
  "examples": [
    {
      "command": "nx g app myapp --directory=myorg",
      "description": "Generate `apps/myorg/myapp` and `apps/myorg/myapp-e2e`"
    },
    {
      "command": "nx g app myapp --classComponent",
      "description": "Use class components instead of functional components"
    }
  ],
  "type": "object",
  "properties": {
    "project": {
      "description": "The root directory of the Cloudscape application.",
      "type": "string",
      "alias": "dir",
      "x-priority": "important",
      "x-prompt": "the root directory for your Cloudscape application",
      "x-dropdown": "projects"
    },
    "cognitoDomain": {
      "description": "The cognito domain prefix to use",
      "type": "string",
      "x-priority": "important",
      "x-prompt": "the cognito domain prefix to use i.e: https://{{cognitoDomain}}-123456789.auth.ap-southeast-2.amazoncognito.com"
    },
    "allowSignup": {
      "description": "Whether to allow self-signup",
      "type": "boolean",
      "default": false,
      "x-prompt": "Enable self signup?",
      "x-priority": "important"
    }
  },
  "required": ["project", "cognitoDomain"]
}
