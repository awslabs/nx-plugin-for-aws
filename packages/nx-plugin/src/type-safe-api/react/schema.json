{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "type-safe-api-react",
  "title": "Integrate a Type Safe API with a React website",
  "type": "object",
  "properties": {
    "hooksLibraryProject": {
      "type": "string",
      "description": "The name of the Type Safe API generated react hooks library project",
      "x-prompt": "The react hooks library project for the Type Safe API to integrate",
      "x-dropdown": "projects"
    },
    "frontendProject": {
      "type": "string",
      "description": "The name of the React project to add the Type Safe API integration to",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "React application's project name.",
      "x-dropdown": "projects"
    },
    "auth": {
      "type": "string",
      "description": "",
      "x-prompt": "Select your API authentication strategy",
      "x-dropdown": ["IAM", "None"],
      "default": "IAM"
    }
  },
  "required": ["hooksLibraryProject", "frontendProject", "auth"]
}
