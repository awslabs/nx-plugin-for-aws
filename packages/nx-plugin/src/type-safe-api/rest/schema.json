{
  "$schema": "https://json-schema.org/schema",
  "$id": "TypeSafeApiModel",
  "title": "Create a Type Safe REST API",
  "description": "Create a Type Safe REST API",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Name of the API.",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What would you like your API to be called? e.g. MyApi",
      "x-priority": "important"
    },
    "modelLanguage": {
      "description": "The language you would like to model your api in.",
      "type": "string",
      "enum": ["openapi", "typespec"],
      "default": "typespec",
      "x-prompt": "What language would you like to use to model your API?",
      "x-priority": "important"
    },
    "infrastructureLanguage": {
      "description": "The language you would like to write your infrastructure in.",
      "type": "string",
      "enum": ["typescript"],
      "default": "typescript",
      "x-prompt": "What language are you using to write CDK infrastructure?",
      "x-priority": "important"
    },
    "handlerLanguages": {
      "description": "The languages you would like to write lambda handlers in.",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["typescript"]
      },
      "default": ["typescript"],
      "x-prompt": "What languages (if any) would you like to write lambda handlers in?",
      "x-priority": "important"
    },
    "libraries": {
      "description": "The libraries to generate from your API",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["typescript-react-hooks"]
      },
      "default": ["typescript-react-hooks"],
      "x-prompt": "What libraries would you like to generate?",
      "x-priority": "important"
    },
    "runtimeLanguages": {
      "description": "The languages you would like to write client or server code in, in addition to the infrastructure or handler languages.",
      "type": "array",
      "items": {
        "type": "string",
        "enum": ["typescript"]
      },
      "default": [],
      "x-prompt": "What other languages (if any) would you like to write client or server code in?"
    },
    "directory": {
      "type": "string",
      "description": "Parent directory where the api model is placed.",
      "default": "packages",
      "x-prompt": "Which directory do you want to create the project in?"
    },
    "scope": {
      "type": "string",
      "description": "Scope for API packages, e.g. @my-company. If omitted this will be inferred",
      "pattern": "^@[a-z0-9-]+(?:/[a-z0-9-]+)?$",
      "x-prompt": "What namespace would you like your library to have? i.e: @myapi"
    },
    "subDirectory": {
      "type": "string",
      "description": "The sub directory the API is placed in. By default this is the API name.",
      "x-prompt": "Which sub directory do you want to create the API in? (By default this is the API name)"
    }
  },
  "required": ["name", "modelLanguage", "infrastructureLanguage"]
}