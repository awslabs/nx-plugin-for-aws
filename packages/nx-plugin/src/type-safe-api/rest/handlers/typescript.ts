/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
import { readProjectConfiguration, Tree, updateProjectConfiguration } from "@nx/devkit";
import { InferredTypeSafeRestApiSchema } from "../schema";
import { buildGenerateTarget } from "../../utils/generate-target";
import { _tsLibGenerator } from "../../../ts/lib/generator";
import { updateGitIgnore } from "../../../utils/gitignore";
import { toScopeAlias } from "../../../utils/npm-scope";
import { join } from "path";
import { addGeneratedExplicitTypeScriptDependency } from "../../utils/generated-ts-lib";


export const typeScriptRestHandlersGenerator = async (tree: Tree, schema: InferredTypeSafeRestApiSchema) => {
  const { dir, fullyQualifiedName, assetPath: bundleOutputDir } = schema.handlers.typescript!;

  await _tsLibGenerator(tree, {
    ...schema.handlers.typescript!,
    unitTestRunner: 'vitest',
    linter: 'eslint',
  });

  if (tree.exists(join(dir, 'src', 'index.ts'))) {
    tree.write(join(dir, 'src', 'index.ts'), '');
  }

  const projectConfig = readProjectConfiguration(tree, fullyQualifiedName);
  updateProjectConfiguration(tree, fullyQualifiedName, {
    ...projectConfig,
    metadata: {
      ...projectConfig.metadata,
      apiName: schema.name,
    } as unknown,
    targets: {
      ...projectConfig.targets,
      generate: buildGenerateTarget({
        modelFullyQualifiedName: schema.model.fullyQualifiedName,
        templateDirs: ['typescript-lambda-handlers'],
        specPath: schema.model.outputSpecPath,
        outputPath: dir,
        metadata: {
          srcDir: 'src',
          tstDir: 'src', // vitest config is all set up for tests co-existing with source
          runtimePackageName: toScopeAlias(schema.runtime.typescript!.fullyQualifiedName),
          esm: true,
          vitest: true,
        },
      }),
      compile: {
        ...projectConfig.targets?.compile,
        dependsOn: ['generate', ...(projectConfig.targets?.compile?.dependsOn ?? [])],
      },
      test: {
        ...projectConfig.targets?.test,
        dependsOn: ['generate', ...(projectConfig.targets?.test?.dependsOn ?? [])],
      },
      bundle: {
        executor: "nx:run-commands",
        options: {
          commands: [
            `rm -rf "${bundleOutputDir}" && mkdir -p "${bundleOutputDir}"`,
            `esbuild --bundle "${join(dir, "src", "*.ts")}" --platform=node --outdir="${bundleOutputDir}" --target=node18`,
            `for f in $(ls "${bundleOutputDir}"); do mkdir "${bundleOutputDir}/$(basename $f .js)" && mv "${bundleOutputDir}/$f" "${bundleOutputDir}/$(basename $f .js)/index.js"; done`,
          ],
          parallel: false,
        },
        dependsOn: ["compile"],
      },
      build: {
        ...projectConfig.targets?.build,
        dependsOn: ['bundle', ...(projectConfig.targets?.build?.dependsOn ?? [])],
      },
    },
  });

  addGeneratedExplicitTypeScriptDependency(tree, schema.handlers.typescript!, schema.runtime.typescript!);

  updateGitIgnore(tree, dir, (patterns) => [
    ...patterns,
    ".tsapi-manifest",
  ]);
};
