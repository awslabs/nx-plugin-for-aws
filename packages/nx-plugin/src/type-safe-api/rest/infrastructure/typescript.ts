/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
import { addDependenciesToPackageJson, readProjectConfiguration, Tree, updateProjectConfiguration } from "@nx/devkit";
import { InferredTypeSafeRestApiSchema } from "../schema";
import { buildGenerateTarget } from "../../utils/generate-target";
import { withVersions } from "../../../utils/versions";
import { toScopeAlias } from "../../../utils/npm-scope";
import { addGeneratedExplicitTypeScriptDependency, generatedTsLibGenerator } from "../../utils/generated-ts-lib";
import { getRelativePathWithinTree } from "../../../utils/paths";
import { join } from "path";
import { updateGitIgnore } from "../../../utils/gitignore";

export const typeScriptRestInfrastructureGenerator = async (tree: Tree, schema: InferredTypeSafeRestApiSchema) => {
  const { dir, fullyQualifiedName } = schema.infrastructure.typescript!;

  await generatedTsLibGenerator(tree, {
    ...schema.infrastructure.typescript!,
  });

  const projectConfig = readProjectConfiguration(tree, fullyQualifiedName);
  updateProjectConfiguration(tree, fullyQualifiedName, {
    ...projectConfig,
    metadata: {
      ...projectConfig.metadata,
      apiName: schema.name,
    } as unknown,
    targets: {
      ...projectConfig.targets,
      generate: buildGenerateTarget({
        modelFullyQualifiedName: schema.model.fullyQualifiedName,
        templateDirs: ['typescript-cdk-infrastructure'],
        specPath: schema.model.outputSpecPath,
        outputPath: dir,
        cacheOutputs: ["mocks"],
        metadata: {
          srcDir: 'src',
          runtimePackageName: toScopeAlias(schema.runtime.typescript!.fullyQualifiedName),
          relativeSpecPath: join("..", getRelativePathWithinTree(tree, dir, schema.model.outputSpecPath)),
          enableMockIntegrations: true,
          ...(schema.handlers.typescript ? {
            "x-handlers-typescript-asset-path": getRelativePathWithinTree(tree, dir, schema.handlers.typescript.assetPath),
          } : {}),
          ...(schema.handlers?.python ? {
            "x-handlers-python-asset-path": getRelativePathWithinTree(tree, dir, schema.handlers.python.assetPath),
            "x-handlers-python-module": schema.handlers.python.moduleName,
          } : {}),
          ...(schema.handlers?.java ? {
            "x-handlers-java-asset-path": getRelativePathWithinTree(tree, dir, schema.handlers.java.assetPath),
            "x-handlers-java-package": schema.handlers.java.packageName,
          } : {}),
          "x-handlers-node-lambda-runtime-version": "NODEJS_20_X",
          "x-handlers-python-lambda-runtime-version": "PYTHON_3_12",
          "x-handlers-java-lambda-runtime-version": "JAVA_21",
          esm: true,
        },
      }),
      mocks: {
        executor: 'nx:run-commands',
        options: {
          parallel: false,
          commands: [
            `type-safe-api generate-mock-data --specPath "${schema.model.outputSpecPath}" --outputPath "${dir}"`,
          ],
        },
        dependsOn: [{
          projects: [schema.model.fullyQualifiedName], target: "build",
        }],
      },
      compile: {
        ...projectConfig.targets?.compile,
        dependsOn: ['generate', ...(projectConfig.targets?.compile?.dependsOn ?? [])],
      },
      build: {
        ...projectConfig.targets?.build,
        dependsOn: ['mocks', ...(projectConfig.targets?.build?.dependsOn ?? [])],
      },
    },
    implicitDependencies: [...(projectConfig.implicitDependencies ?? []), schema.model.fullyQualifiedName],
  });

  addGeneratedExplicitTypeScriptDependency(tree, schema.infrastructure.typescript!, schema.runtime.typescript!);

  updateGitIgnore(tree, dir, (patterns) => [
    ...patterns,
    "mocks",
  ]);

  addDependenciesToPackageJson(tree, withVersions([
    "@aws/pdk",
    "aws-cdk-lib",
    "constructs",
  ]), {});
};
