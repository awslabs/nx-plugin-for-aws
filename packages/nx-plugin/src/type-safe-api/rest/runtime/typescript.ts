/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
import { addDependenciesToPackageJson, readProjectConfiguration, Tree, updateProjectConfiguration } from "@nx/devkit";
import { InferredTypeSafeRestApiSchema } from "../schema";
import { buildGenerateTarget } from "../../utils/generate-target";
import { withVersions } from "../../../utils/versions";
import { generatedTsLibGenerator } from "../../utils/generated-ts-lib";


export const typeScriptRestRuntimeGenerator = async (tree: Tree, schema: InferredTypeSafeRestApiSchema) => {
  const { dir, fullyQualifiedName } = schema.runtime.typescript!;

  await generatedTsLibGenerator(tree, {
    ...schema.runtime.typescript!,
  });

  const projectConfig = readProjectConfiguration(tree, fullyQualifiedName);
  updateProjectConfiguration(tree, fullyQualifiedName, {
    ...projectConfig,
    metadata: {
      ...projectConfig.metadata,
      apiName: schema.name,
    } as unknown,
    targets: {
      ...projectConfig.targets,
      generate: buildGenerateTarget({
        modelFullyQualifiedName: schema.model.fullyQualifiedName,
        templateDirs: ['typescript'],
        specPath: schema.model.outputSpecPath,
        outputPath: dir,
        metadata: {
          srcDir: 'src',
          esm: true,
        },
      }),
      compile: {
        ...projectConfig.targets?.compile,
        dependsOn: ['generate', ...(projectConfig.targets?.compile?.dependsOn ?? [])],
      },
    },
    implicitDependencies: [...(projectConfig.implicitDependencies ?? []), schema.model.fullyQualifiedName],
  });

  addDependenciesToPackageJson(tree, withVersions([
    "@aws-lambda-powertools/tracer",
    "@aws-lambda-powertools/logger",
    "@aws-lambda-powertools/metrics",
  ]), withVersions([
    "@types/aws-lambda",
  ]));
};
