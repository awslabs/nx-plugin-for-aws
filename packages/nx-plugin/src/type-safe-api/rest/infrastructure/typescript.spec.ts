/**
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { Tree } from '@nx/devkit';
import { typeScriptRestInfrastructureGenerator } from './typescript';
import { InferredTypeSafeRestApiSchema } from '../schema';

describe('type-safe-api rest typescript infrastructure generator', () => {
  let tree: Tree;
  let options: InferredTypeSafeRestApiSchema;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
    options = {
      name: 'test-api',
      modelLanguage: 'openapi',
      infrastructureLanguage: 'typescript',
      handlerLanguages: ['typescript'],
      libraries: [],
      runtimeLanguages: ['typescript'],
      fullyQualifiedApiName: 'test-api',
      nameKebabCase: 'test-api',
      namePascalCase: 'TestApi',
      dir: 'apis/test-api',
      model: {
        dir: 'apis/test-api-model',
        fullyQualifiedName: 'test-api-model',
        outputSpecPath: 'apis/test-api-model/dist/spec.json'
      },
      runtime: {
        dir: 'apis/test-api-runtime',
        typescript: {
          dir: 'apis/test-api-runtime/typescript',
          fullyQualifiedName: 'test-api-runtime-typescript'
        }
      },
      infrastructure: {
        dir: 'apis/test-api-infrastructure',
        typescript: {
          dir: 'apis/test-api-infrastructure/typescript',
          fullyQualifiedName: 'test-api-infrastructure-typescript'
        }
      },
      handlers: {
        dir: 'apis/test-api-handlers',
        typescript: {
          dir: 'apis/test-api-handlers/typescript',
          fullyQualifiedName: 'test-api-handlers-typescript',
          assetPath: 'apis/test-api-handlers/typescript/dist'
        }
      },
      library: {
        dir: 'apis/test-api-library'
      }
    };
  });

  it('should configure project.json with appropriate targets', async () => {
    await typeScriptRestInfrastructureGenerator(tree, options);

    const projectConfig = JSON.parse(
      tree.read('apis/test-api-infrastructure/typescript/project.json', 'utf-8')
    );

    // Check metadata
    expect(projectConfig.metadata.apiName).toBe('test-api');

    // Check generate target configuration
    expect(projectConfig.targets.generate).toBeDefined();
    expect(projectConfig.targets.generate.executor).toBe('nx:run-commands');
    expect(projectConfig.targets.generate.options.commands[0]).toContain('type-safe-api generate');
    expect(projectConfig.targets.generate.options.commands[0]).toContain('--specPath "apis/test-api-model/dist/spec.json"');
    expect(projectConfig.targets.generate.options.commands[0]).toContain('--outputPath "apis/test-api-infrastructure/typescript"');
    expect(projectConfig.targets.generate.options.commands[0]).toContain('--templateDirs "typescript-cdk-infrastructure"');

    // Check mocks target configuration
    expect(projectConfig.targets.mocks).toBeDefined();
    expect(projectConfig.targets.mocks.executor).toBe('nx:run-commands');
    expect(projectConfig.targets.mocks.options.commands[0]).toContain('type-safe-api generate-mock-data');
    expect(projectConfig.targets.mocks.dependsOn[0].projects).toContain('test-api-model');

    // Check compile and build target dependencies
    expect(projectConfig.targets.compile.dependsOn).toContain('generate');
    expect(projectConfig.targets.build.dependsOn).toContain('mocks');

    // Snapshot full project.json
    expect(projectConfig).toMatchSnapshot();
  });

  it('should add implicit dependency on model project', async () => {
    await typeScriptRestInfrastructureGenerator(tree, options);

    const projectConfig = JSON.parse(
      tree.read('apis/test-api-infrastructure/typescript/project.json', 'utf-8')
    );

    expect(projectConfig.implicitDependencies).toContain('test-api-model');
  });

  it('should add implicit dependency on runtime project', async () => {
    await typeScriptRestInfrastructureGenerator(tree, options);

    const projectConfig = JSON.parse(
      tree.read('apis/test-api-infrastructure/typescript/project.json', 'utf-8')
    );

    expect(projectConfig.implicitDependencies).toContain('test-api-runtime-typescript');
  });

  it('should add required dependencies to package.json', async () => {
    await typeScriptRestInfrastructureGenerator(tree, options);

    const packageJson = JSON.parse(tree.read('package.json', 'utf-8'));

    // Check dependencies are added
    expect(packageJson.dependencies['@aws/pdk']).toBeDefined();
    expect(packageJson.dependencies['aws-cdk-lib']).toBeDefined();
    expect(packageJson.dependencies['constructs']).toBeDefined();
  });

  it('should add required entries to .gitignore', async () => {
    await typeScriptRestInfrastructureGenerator(tree, options);

    const gitignore = tree.read('apis/test-api-infrastructure/typescript/.gitignore', 'utf-8');
    expect(gitignore).toContain('mocks');
    expect(gitignore).toContain('src');
    expect(gitignore).toContain('.tsapi-manifest');
  });
});
