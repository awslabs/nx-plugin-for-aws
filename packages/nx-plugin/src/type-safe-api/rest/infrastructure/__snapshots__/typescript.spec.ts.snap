// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`type-safe-api rest typescript infrastructure generator > should configure project.json with appropriate targets 1`] = `
{
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "implicitDependencies": [
    "test-api-model",
    "test-api-runtime-typescript",
  ],
  "metadata": {
    "apiName": "test-api",
  },
  "name": "test-api-infrastructure-typescript",
  "projectType": "library",
  "sourceRoot": "apis/test-api-infrastructure/typescript/src",
  "tags": [],
  "targets": {
    "build": {
      "dependsOn": [
        "mocks",
        "compile",
      ],
    },
    "compile": {
      "dependsOn": [
        "generate",
      ],
      "executor": "@nx/js:tsc",
      "options": {
        "assets": [
          "apis/test-api-infrastructure/typescript/*.md",
        ],
        "main": "apis/test-api-infrastructure/typescript/src/index.ts",
        "outputPath": "dist/apis/test-api-infrastructure/typescript",
        "tsConfig": "apis/test-api-infrastructure/typescript/tsconfig.lib.json",
      },
      "outputs": [
        "{options.outputPath}",
      ],
    },
    "generate": {
      "cache": true,
      "dependsOn": [
        {
          "projects": [
            "test-api-model",
          ],
          "target": "build",
        },
      ],
      "executor": "nx:run-commands",
      "options": {
        "commands": [
          "type-safe-api generate --specPath "apis/test-api-model/dist/spec.json" --outputPath "apis/test-api-infrastructure/typescript" --templateDirs "typescript-cdk-infrastructure" --metadata '{"srcDir":"src","runtimePackageName":"test-api-runtime-typescript","relativeSpecPath":"../../../test-api-model/dist/spec.json","enableMockIntegrations":true,"x-handlers-typescript-asset-path":"../../test-api-handlers/typescript/dist","x-handlers-node-lambda-runtime-version":"NODEJS_20_X","x-handlers-python-lambda-runtime-version":"PYTHON_3_12","x-handlers-java-lambda-runtime-version":"JAVA_21","esm":true}'",
        ],
        "parallel": false,
      },
      "outputs": [
        "{projectRoot}/.tsapi-manifest",
        "{projectRoot}/src",
        "{projectRoot}/mocks",
      ],
    },
    "mocks": {
      "dependsOn": [
        {
          "projects": [
            "test-api-model",
          ],
          "target": "build",
        },
      ],
      "executor": "nx:run-commands",
      "options": {
        "commands": [
          "type-safe-api generate-mock-data --specPath "apis/test-api-model/dist/spec.json" --outputPath "apis/test-api-infrastructure/typescript"",
        ],
        "parallel": false,
      },
    },
  },
}
`;
